src/app/data/services/local_storage_service.ts

```diff
@@ -0,0 +1,26 @@
+///<reference path="../models/content_block.ts"/>
+
+class LocalStorageService {
+  static EXPIRATIONTIME = 5; //minutes
+  private expiryDate:{[key: string]: Date} = {};
+
+  fetch(key:string):AjaxContentBlocksHash {
+    return JSON.parse(localStorage.getItem(key));
+  }
+
+  put(key:string, hash:AjaxContentBlocksHash) {
+    if (this.isValid(key)) return;
+    this.setExpiryDate(key);
+    localStorage.setItem(key, JSON.stringify(hash));
+  }
+
+  private isValid(key:string) {
+    return <any>this.expiryDate[key] - Date.now() > 0;
+  }
+
+  private setExpiryDate(key:string) {
```

Comment:
maybe we should use Date.now(), which returns the current milliseconds https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

This way we don't have to use the date object. E.g.

```js
static EXPIRATION_TIME = 5 * 60 * 1000; // 5 minutes

this.expiryDate[key] = Date.now() + LocalStorageService.EXPIRATION_TIME;
```